### PR TITLE
Versioning with poetry and automatic release

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Use poetry to manage versioning.
Use semantic versioning and autobump on push to main, but only bump minor version.
A bot commit will be pushed to bump the version in `pyproject.toml`

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -59,3 +59,19 @@ markers = [
 
 [tool.pylint]
 disable = "wrong-import-order"
+
+[tool.poetry]
+name = "flask-multipass-saml-groups"
+version = "0.1.0"
+description = "This package provides an identity provider for Flask-Multipass, which allows you to use SAML groups. It is designed to be used as a plugin for Indico."
+authors = ["Your Name <you@example.com>"]
```
Comment (cbartz) [inline pyproject.toml:67]:
```suggestion
authors = ["launchpad.net/~canonical-is-devops <is-devops-team@canonical.com>"]
```

File: .github/workflows/create_release.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                echo VERSION=${VERSION} >> $GITHUB_ENV
+
+                # Tag the commit with the new version
+                git tag $VERSION
+                git push origin $VERSION
+
+                # Push version update in pyproject.toml
+                git config --global user.name 'IS DevOps Bot'
+                git config --global user.email 'is-devops-bot@users.noreply.github.com'
+                git add .
```
Comment (cbartz) [inline .github/workflows/create_release.yaml:39]: Is it safe to add all files, or are there some that are better ignored? poetry would create a lock file e.g., which might be fine to add.

Comment (Thanhphan1147) [reply .github/workflows/create_release.yaml]: poetry.lock won't be generated here but I changed the command to only commit pyproject.toml

File: pyproject.toml
[the remark this answers is not in the document]
Comment (Thanhphan1147) [reply pyproject.toml:1]: Thanks for the reminder. I just added dependencies versioning in `pyproject.toml` and also updated tox.ini to reflect this. So I removed `requirements.txt`

**Review by cbartz — DISMISSED** [review]

**Review by cbartz — DISMISSED** [review]

**Review by merkata — DISMISSED** [review]
looks good, integ is failing (should be libjuju related)

**Review by javierdelapuente — APPROVED** [review]

**Review by cbartz — APPROVED** [review]

**Pull request merged**